exonum/src/helpers/merkle.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Merkle tree utilities.
+
+use crypto::{self, Hash, HashStream};
+
+/// Computes Merkle root hash for a given list of hashes.
+///
+/// If `hashes` are empty then `Hash::zero()` value is returned.
+pub fn root_hash<'a, I: IntoIterator<Item = &'a Hash>>(hashes: I) -> Hash {
+    let mut current_hashes: Vec<Hash> = hashes.into_iter().cloned().collect();
+    if current_hashes.is_empty() {
+        return Hash::zero();
+    }
+    while current_hashes.len() > 1 {
+        current_hashes = combine_hash_list(&current_hashes);
+    }
+    current_hashes[0]
+}
+
+fn combine_hash_list(hashes: &[Hash]) -> Vec<Hash> {
+    hashes
+        .chunks(2)
+        .map(|pair|
+            // Keep hash combination consistent with ProofListIndex.
+            if pair.len() == 2 {
+                hash_pair(&pair[0], &pair[1])
+            } else {
+                hash_one(&pair[0])
+            }
+        )
+        .collect()
+}
+
+fn hash_one(lhs: &Hash) -> Hash {
```

Comment:
Would it be okay to move `hash_{one,pair}` to `proof_list_index` module and export them as `pub(crate)` so that they are visible to `helpers::merkle`? Or it would be better to just move `root_hash()` from `helpers::merkle` to `proof_list_index`?